core/src/test/java/org/elasticsearch/index/seqno/GlobalCheckpointTrackerTests.java

```diff
@@ -523,200 +540,21 @@ public void testRaceUpdatingGlobalCheckpoint() throws InterruptedException, Brok
         assertThat(tracker.getGlobalCheckpoint(), equalTo((long) nextActiveLocalCheckpoint));
     }
 
-    public void testPrimaryContextOlderThanAppliedClusterState() {
```

Comment:
I think we need equivalent tests of `initializeWithPrimaryContext` and it's relation to `appliedClusterStateVersion` ?